test/load_ctl_SUITE.erl

```diff
@@ -336,27 +356,98 @@ lc_alarm({init, Config}) ->
   Config;
 lc_alarm(Config) ->
   alarm_handler:start_link(),
-  lc_flagman_recover(Config),
-  [{?LC_ALARM_ID_RUNQ, #{node := Node, runq_length := QLen}}] = alarm_handler:get_alarms(),
-  ?assertEqual(Node, node()),
+  lc_runq_recover(Config),
+  #{node := Node, runq_length := QLen} =
+    proplists:get_value(?LC_ALARM_ID_RUNQ, alarm_handler:get_alarms()),
+  ?assertEqual(node(), Node),
   ?assert(QLen > 0),
   LConfig = load_ctl:get_config(),
   load_ctl:put_config(LConfig#{?RUNQ_MON_F2 => 0.5}),
   timer:sleep(5000),
-  ?assertMatch([], alarm_handler:get_alarms()),
+  ct:pal("checking_alarm..."),
+  ?assertEqual(undefined, proplists:get_value(?LC_ALARM_ID_RUNQ,
+                                              alarm_handler:get_alarms())),
   ok.
 
 lc_alarm2({init, Config}) ->
   Config;
 lc_alarm2(Config) ->
   alarm_handler:start_link(),
-  lc_flagman_recover(Config),
-  ?assertMatch([{?LC_ALARM_ID_RUNQ, _}], alarm_handler:get_alarms()),
+  lc_runq_recover(Config),
+  ?assertMatch(#{node := _, runq_length := _},
+               proplists:get_value(?LC_ALARM_ID_RUNQ, alarm_handler:get_alarms())),
   ok = load_ctl:stop_runq_flagman(10000),
   timer:sleep(100),
-  ?assertMatch([], alarm_handler:get_alarms()),
+  ?assertEqual(undefined,
+               proplists:get_value(?LC_ALARM_ID_RUNQ, alarm_handler:get_alarms())),
   ok.
 
+lc_mem({init, Config})->
+  Config;
+lc_mem(_Config) ->
+  ?assert(lc_lib:get_memory_usage() < ?MEM_MON_F1_DEFAULT),
+  ?check_trace(#{timetrap => 30000},
+               begin
+                 meck:new(lc_lib, [passthrough]),
+                 meck:expect(lc_lib, get_memory_usage,
+                             fun() -> ?MEM_MON_F1_DEFAULT + 0.01 end
+                            ),
+                 timer:sleep(?MEM_MON_T1_DEFAULT*2),
+                 Check1 = load_ctl:is_high_mem(),
+                 ?assert(load_ctl:is_high_mem()),
+                 meck:unload(lc_lib),
+                 timer:sleep(?MEM_MON_T1_DEFAULT*2),
+                 Check2 = load_ctl:is_high_mem(),
+                 {Check1, Check2}
+               end,
+               fun(Result, Trace) ->
+                   ct:pal("Trace is ~p", [Trace]),
+                   ?assertEqual({true, false}, Result),
+                   ?assert(?strict_causality(
+                              #{'$kind' := lc_flag_man
+                               , event := raise_flag},
+                              #{'$kind' := lc_flag_man
```

Comment:
```suggestion
                              #{?snk_kind := lc_flag_man
```
?

Reply:
yes. It was just easier when I write the test by coping it from the trace log. 😄